File: HandBallStats_api/src/main/java/es/diego/handballstats_api/models/Partido.java
package es.diego.handballstats_api.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import es.diego.handballstats_api.models.estadisticas.Estadistica;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name ="partido")
public class Partido {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column(name = "nombre", nullable = false)
    private String nombre;

    @Column(name = "goles_favor")
    private int goles_favor;

    @Column(name = "goles_contra")
    private int goles_contra;

    @Column(name = "fecha", nullable = false)
    private Date fecha;

    @ManyToOne
    @JoinColumn(name = "id_equipo")
    @JsonBackReference("c")
    private Equipo equipo;

    @OneToMany(mappedBy = "partido", cascade = CascadeType.ALL)
    @JsonManagedReference("d")
    private List<Estadistica> estadisticas;
}

File: HandBallStats/app/src/main/java/es/diego/handballstats/models/Partido.kt
package es.diego.handballstats.models

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class Partido(
    var id: Int?,
    val nombre: String,
    val goles_favor: Int,
    val goles_contra: Int,
    val equipo: Equipo,
    val estadisticas: List<Estadistica>? = listOf(),
    val fecha: String = SimpleDateFormat("yyyy-MM-dd", Locale.getDefault()).format(Date())
)